source/Irrlicht/COpenGLExtensionHandler.cpp

```diff
@@ -977,7 +977,8 @@ void COpenGLExtensionHandler::initExtensions(bool stencilBuffer)
 
     num=0;
     glGetIntegerv(GL_MAX_COMBINED_TEXTURE_IMAGE_UNITS, &num);
-	MaxTextureUnits = core::min_(static_cast<uint8_t>(num), static_cast<uint8_t>(MATERIAL_MAX_TEXTURES));
+    //can we get rid of that? I don't really see reason for this clamp at this level of abstraction
```

Comment:
SMaterial only has `MATERIAL_MAX_TEXTURES` slots for now, once we solve #129 #130 we can remove this limit.